clients/drcachesim/reader/reader.cpp

```diff
@@ -386,6 +386,13 @@ reader_t::process_input_entry()
             this, 2,
             "Assuming header is part of concatenated or on-disk-core-sharded traces\n");
         break;
+    case TRACE_TYPE_FOOTER:
+        // We support core-sharded-on-disk traces where an originally-thread-sharded
+        // input ends but the core-sharded new trace continues.
+        VPRINT(
```

Comment:
Prior to this PR, I assume TRACE_TYPE_FOOTER entries were falling to the default case, and failing due to the `assert(false)`? What about the L93 if-block for TRACE_TYPE_FOOTER in operator++, which should avoid the process_input_entry call for TRACE_TYPE_FOOTER? Or maybe this affected only when process_input_entry was invoked in one of the skip-related logic?

Reply:
It is the latter: the skip code calling process_input_entry.